### ISSUE TITLE
401 upon logging in again using keycloak SSO and rootpath option

### ISSUE THREAD

**Issue by lanmarti** [argoproj/argo-cd#3552]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

After logging out of the web ui, when trying to log in again, the `GET api/v1/clusters` and `GET api/v1/applications?fields=items.metadata.name%2Citems.metadata.annotations%2Citems.metadata.labels%2Citems.metadata.resourceVersion%2Citems.metadata.creationTimestamp%2Citems.metadata.deletionTimestamp%2Citems.spec%2Citems.operation.sync%2Citems.status.sync.status%2Citems.status.health%2Citems.status.operationState.phase%2Citems.status.operationState.operation.sync%2Citems.status.summary&selector=` calls return a 401 and I get redirected back to the login screen.
Clearing the browser cache solves the issue, clearing active sessions in Keycloak does not.

The failing call seems to contain a valid token, i.e. not expired, contains the required groups field etc. The only visible difference is that the first call contains a nonce, while the second does not (id token vs access token perhaps?). 
However, if I clear active client sessions through Keycloak without clearing browser cache, the token contains a nonce but still results in a 401 and redirects back to login page.

**To Reproduce**

Set up SSO using a Keycloak OIDC client
Log in using SSO
Log out
Log in again

**Expected behavior**

The second login should succeed as well.

**Version**

```shell
argocd: v1.5.4+36bade7
  BuildDate: 2020-05-05T18:59:08Z
  GitCommit: 36bade7a2d7b69d1c0b0c4d41191f792a847d61c
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
argocd-server: v1.5.3+095c5d6
  BuildDate: 2020-05-02T04:21:47Z
  GitCommit: 095c5d616b1cb39f87e8f7a54cabea2643e1c99a
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: {Version:kustomize/v3.5.4 GitCommit:3af514fa9f85430f0c1557c4a0291e62112ab026 BuildDate:2020-01-11T03:12:59Z GoOs:linux GoArch:amd64}
  Helm Version: version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
  Kubectl Version: v1.14.0
```

**Comment by lanmarti** [comment]
After some searching, I've found that the failing calls actually contain 'argocd.token' twice in the 'Cookie' header.
argocd.token=; argocd.token=[valid token]
I'm guessing this is causing the issue, but I have no clue if this is caused by keycloak, argocd or any of the network components in between...

**Comment by lanmarti** [comment]
After some more searching, I've found that the problem only occurs when I'm using the new --rootpath option introduced in v1.5.

**Comment by lanmarti** [comment]
This issue seems to have been fixed after upgrading to v1.5.8, I suspect through #3632
